neuro_san/client/thinking_file_message_processor.py

```diff
@@ -89,31 +93,68 @@ def write_message(self, response: Dict[str, Any], origin_str: str):
             # There is no real text, but there is a structure. JSON-ify it.
             text = json.dumps(structure, indent=4, sort_keys=True)
 
+        # Figure out how we are going to report the origin given the message.
+        use_origin: str = self._determine_origin_reporting(response, origin_str)
+
+        # Determine the filename to use given the origin_str.
+        # Previously we might have used a uuid as a filename, but by default
+        # we want the filenames to match the full origin_str.
+        origin_filename: str = self.origins.get(origin_str, origin_str)
+
+        try:
+            self._write_to_file(origin_filename, origin_str, message_type_str, use_origin, text)
+        except OSError as os_error:
+            # For very deep networks we can sometimes get a "File name too long",
+            # which is Error code 63.
+            if os_error.errno == 63:
+
+                # Retry with a uuid as file name.
+                # If this fails, there's no helping ya.
+                origin_filename = str(uuid.uuid4())
+                self._write_to_file(origin_filename, origin_str, message_type_str, use_origin, text)
+
+                # Squirell that uuid away so results continue to go to the same
+                # file over and over again.
+                self.origins[origin_str] = origin_filename
+            else:
+                raise os_error
```

Comment:
Do normal write to file but if the write fails because of Filename too long, try again with a uuid as filename.